### PR TITLE
Side bar

### PR DESCRIPTION
**Pull request by shushengfuqin**
### Summary 
This is just a pr which contains the basic part of the new UI.
They are in the new folder under src called UIComponent. 
Please becareful with the naming convention and how the files are placed.

### Details
Action bar: the bar on the left which contains the buttons to change the content in the sidebar
Side bar: the togglebal container which contains different element based on the button clicked in the action bar
Textbox : the text box
ActionSequence"
### How to test?
There not much to be tested, However if you need to work on the unity and have the previous things, just go in the webglbox.tsx and uncomment the overlay and comment the ActionBar.
### Checks 
- [ ] Tested Changes
- [ ] Stakeholder Approval

### CODE REVIEW

File: src/util/customHooks/ActionBarFunc.tsx [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+  const [tooDoc, setToolDoc] = useState(true);
+  const toggleToolDoc = () => {
+    setToolDoc((v) => !v);
+    setApparatusInfo(false);
+  };
+
+  // hook for showing or not action list
+  const [apparatusInfo, setApparatusInfo] = useState(true);
+  const toggleApparatusInfo = () => {
+    if (tooDoc === false) {
```
Comment (Ysabz) [inline src/util/customHooks/ActionBarFunc.tsx:14]: typo

**Review by Ysabz — CHANGES_REQUESTED** [review]
Please put all the new UI components in a separate folder for now for a cleaner folder structure

**Comment by Ysabz** [comment]
The page does not fit on the screen ( there are side scrollbars no matter what Zoom I use) 
image

**Comment by Ysabz** [comment]
This PR needs to integrate the new selection page and text box

**Comment by Ysabz** [comment]
> The page does not fit on the screen ( there are side scrollbars no matter what Zoom I use) image

also, in the side bar, what is the second button from top doing?

**Comment by Ysabz** [comment]
problems with action sequence :

- the place holders in the action sequence are not necessary. 
- the color of elements in action sequence need to be consistent with mockup
- The action sequence is not scrollable and draggable 
- not connected with other components

File: src/components/UIComponent/SideBarItem.tsx [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+
+const SideBarItemWrapper = styled.div`
+  display: flex;
+  justify-content: center;
+`;
+
+function SideBarItem(): JSX.Element {
+  const { toggleApparatusInfo, toggleToolDoc } = useContext(SideBarContext);
+  const { userId, experienceData } = useContext(experienceContext);
+  const { actionList } = useActionList(experienceData);
```
[the remark this answers is not in the document]
Comment (shushengfuqin) [reply src/components/UIComponent/SideBarItem.tsx:37]: True, 
I'll make it as context in this case.

File: src/components/UIComponent/Action Sequence 2.0/ActionSequenceItem.tsx [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+  justify-content: center;
+  margin: 0.5em;
+`;
+
+type ActionSequenceItemProps = {
+  name:string
+  action:string
+}
+
+function ActionSequenceItem({name,action}:ActionSequenceItemProps): JSX.Element {
```
[the remark this answers is not in the document]
Comment (shushengfuqin) [reply src/components/UIComponent/Action Sequence 2.0/ActionSequenceItem.tsx:36]: that i have no idea, it was merged from what Justin did. I'll double-check to make sure.

File: src/components/UIComponent/Action Sequence 2.0/ActionSequence.tsx [hunk 4]
```diff
@@ -0,0 +1,108 @@
...
+function ActionSequence({
+  actionList,
+  removeAction,
+  handleOnDragEnd,
+}: ActionSequenceProps): JSX.Element {
+  return (
+    <ActionSequenceRoot>
+      <ActionSequenceHeader />
+      <ActionSequenceList>
+      {/* <DragDropContext onDragEnd={handleOnDragEnd}>
```
[the remark this answers is not in the document]
Comment (shushengfuqin) [reply src/components/UIComponent/Action Sequence 2.0/ActionSequence.tsx:49]: It was commented by justin, I am not sure why.

File: src/util/customHooks/ActionBarFunc.tsx (hunk 1 above)
Comment (shushengfuqin) [reply src/util/customHooks/ActionBarFunc.tsx:14]: resolved

File: src/components/UIComponent/SideBar.tsx [hunk 5]
```diff
@@ -0,0 +1,132 @@
...
+type OverlayProps = {
+  userId: string;
+  experienceData: ExperienceData;
+};
+
+function SideBar({ userId, experienceData }: OverlayProps): JSX.Element {
+  const {
+    toggleToolDoc,
+    toggleApparatusInfo,
+    tooDoc,
```
[the remark this answers is not in the document]
Comment (shushengfuqin) [reply src/components/UIComponent/SideBar.tsx:73]: tooDoc -> toolDoc
resolved

File: package.json [hunk 6]
```diff
@@ -18,9 +18,12 @@
     "@azure/storage-blob": "^12.8.0",
     "@emotion/react": "^11.4.1",
     "@emotion/styled": "^11.3.0",
+    "@material-ui/core": "^4.12.3",
```
[the remark this answers is not in the document]
Comment (shushengfuqin) [reply package.json:21]: will be removed

File: src/util/customHooks/SideBarContext.tsx [hunk 7]
```diff
@@ -0,0 +1,3 @@
+import { createContext } from "react";
```
[the remark this answers is not in the document]
Comment (shushengfuqin) [reply src/util/customHooks/SideBarContext.tsx:1]: easier to manage.

File: src/components/UIComponent/Action Sequence 2.0/ActionSequence.tsx [hunk 8]
```diff
@@ -0,0 +1,114 @@
...
+                    {(dragProvided) => (
+                      <DragContainer
+                        {...dragProvided.draggableProps}
+                        {...dragProvided.dragHandleProps}
+                        ref={dragProvided.innerRef}
+                      >
+
+
+
+                        {/* <List>
```
[the remark this answers is not in the document]
Comment (shushengfuqin) [reply src/components/UIComponent/Action Sequence 2.0/ActionSequence.tsx:75]: i'm not sure if justin need them or not

File: src/components/UIComponent/ApparatusInfo.tsx [hunk 9]
```diff
@@ -0,0 +1,39 @@
...
+import { experienceContext } from "../../util/customHooks/experienceContext";
+import { SerializedApparatus } from "../../util/types";
+import ApparatusListBox from "../editorBoxes/ApparatusListBox";
+import { useActionList } from "../../util/customHooks/overlayfunc";
+import ActionBox from "../editorBoxes/ActionBox";
+
+function ApparatusInfo(): JSX.Element {
+  const { experienceData } = useContext(experienceContext);
+  const { addActionToList } =
+    useActionList(experienceData);
```
[the remark this answers is not in the document]
Comment (shushengfuqin) [reply src/components/UIComponent/ApparatusInfo.tsx:15]: this is the part that we need useContext i guess? to share the values

**Review by Ysabz — COMMENTED** [review]
Works fine. needs to be enhaned in a another PR

**Pull request merged**